src/core/contracts/registry.ts

```diff
@@ -199,3 +199,35 @@ const getContract = async (opts?: ConfigOpts) => {
 
   return Registry__factory.connect(address, provider);
 };
+
+/**
+ * getRegistrationsByIdentityAddress() Resolves all registrations associated to an identity contract address
+ *
+ * @param identityAddress - The identity contract address to resolve the registrations for
+ * @returns A list of registrations that are associated to the identity contract address
+ */
+export const getRegistrationsByIdentityAddress = async (identityAddress: HexString): Promise<Registration[]> => {
+  const lastRegistryUpdates = await getLastRegistryUpdatesFor(identityAddress);
+
+  const registrationsHandles: Handle[] = lastRegistryUpdates.map((registrations: Registration) => registrations.handle);
+
+  if (registrationsHandles.length === 0) return [];
+
+  const allRegistrations = await Promise.all(registrationsHandles.map((handle: Handle) => resolveRegistration(handle)));
```

Comment:
I'm a little unclear what 210-216 is doing. `getLastRegistryUpdatesFor` already returns the registrations, right? Why do we need to pull the handles out of them and re-resolve those handles to registrations?

Reply:
DSNPRegistryUpdate is triggered for
    -  New registrations
      - emit DSNPRegistryUpdate(newReg.id, newReg.identityAddress, newHandle)
    - Handle updates (changeHandle)
      - emit DSNPRegistryUpdate(newReg.id, newReg.identityAddress, newHandle)
    - Identity contract address changes (changeAddress)
      - emit DSNPRegistryUpdate((reg.id, newAddr, handle);
  
Since DSNPRegistryUpdate is used when updating an identity address we need to make sure that this registration is associated to this address.